cardano-cli/src/Cardano/CLI/Read.hs

```diff
@@ -617,23 +622,19 @@ instance Error CddlError where
       prettyError e
 
 readCddlTx :: FileOrPipe -> IO (Either (FileError TextEnvelopeCddlError) CddlTx)
-readCddlTx = readFileOrPipeTextEnvelopeCddlAnyOf teTypes
+readCddlTx =
+  readFileOrPipeTextEnvelopeCddlAnyOf $
+    map (`FromCDDLTx` CddlTx) txTextEnvelopeTypes
+
+txTextEnvelopeTypes :: [Text]
+txTextEnvelopeTypes =
+  [ let TextEnvelopeType d = shelleyBasedEraConstraints sbe $ textEnvelopeType (proxyToAsType (makeTxProxy sbe))
```

Comment:
Where are those `HasTextEnvelope` and `HasTypeProxy` instances defined? How this mapping works? Previously we had both `Witnessed` and `Unwitnessed` for each era, so this looks like it would support only one for each era.

Reply:
> Where are those HasTextEnvelope and HasTypeProxy instances defined?

`HasTypeProxy` is here: [ [cardano-api] Cadano.Api.Tx.Sign#L184 ](https://github.com/IntersectMBO/cardano-api/blob/7792eb9cfa12c7e9623050f29964383021c17392/cardano-api/internal/Cardano/Api/Tx/Sign.hs#L184)
`HasTextEnvelope` is here: [ [cardano-api] Cadano.Api.Tx.Sign#L281 ](https://github.com/IntersectMBO/cardano-api/blob/7792eb9cfa12c7e9623050f29964383021c17392/cardano-api/internal/Cardano/Api/Tx/Sign.hs#L281)

> How this mapping works? Previously we had both Witnessed and Unwitnessed for each era, so this looks like it would support only one for each era.

We will still support both, but we will use the same type annotation for both since we cannot tell which is which from the type when serialising. We have already been producing them like that, is just we will stop supporting the old format as input. See discussion here: [Slack discussion](https://input-output-rnd.slack.com/archives/G011N23CEAE/p1726503763112519?thread_ts=1725887992.734849&cid=G011N23CEAE)

Serialisation of `TextEnvelopes` is implemented here: [ [cardano-api] Cardano.Api.SerialiseTextEnvelope#L100 ](https://github.com/IntersectMBO/cardano-api/blob/7792eb9cfa12c7e9623050f29964383021c17392/cardano-api/internal/Cardano/Api/SerialiseTextEnvelope.hs#L100)

Reply:
Thanks for the links. Makes sense now.

Reply:
> We will still support both,

We need to remove `legacyComparison` from `cardano-api`.

Reply:
We should consider moving these tests and the tx reading capabilities to `cardano-api`.